### PR TITLE
add `D.StudentT`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by takagi** [comment]
Would you check the CI failure on flake8?

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/distributions/student_t.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    def sample_n(self, n):
+        xp = cuda.get_array_module(self.nu)
+        if xp is cuda.cupy:
+            eps = xp.random.standard_t(
+                df=cuda.to_cpu(self.nu.data),
+                size=(n,)+self.batch_shape, dtype=self.nu.dtype)
+        else:
+            eps = xp.random.standard_t(
+                df=cuda.to_cpu(self.nu.data),
```
Comment (takagi) [inline chainer/distributions/student_t.py]: Do we need `cuda.to_cpu()`?

**Comment by takagi** [comment]
How about adding KL divergence test for this?

**Comment by takagi** [comment]
@toslunar Would you also review this PR because I'm not with expertise in probability and statistics.

File: chainer/distributions/student_t.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+class StudentT(distribution.Distribution):
+
+    """StudentT Distribution.
+
+    The probability density function of the distribution is expressed as
+
+    .. math::
+        p(x) = \\frac{\\Gamma(\\frac{\\nu+1}{2})} \
+            {\\sqrt{\\nu\\pi}\\Gamma(\\frac{\\nu}{2})} \
+            \\left(1 + \\frac{x^2}{\\nu} \\right)^{-(\\frac{\\nu+1}{2})},
```
Comment (toslunar) [inline chainer/distributions/student_t.py]: if loc=0 and scale=1.

**Pull request closed without merging**